docs/old-reference-guide/modules/axon-framework-commands/pages/command-handlers.adoc

```diff
@@ -1,153 +1,215 @@
+    if (cmd.newAmount() > maximumAllowed) {
+        throw new IllegalArgumentException("Amount exceeds maximum");
+    }
+
+    // Publish event
+    appender.append(new CardUpdatedEvent(id, cmd.newAmount()));
+}
+----
+
+The `EventAppender` provides a clean API for publishing events without relying on static methods or thread-local storage.
 
-In some cases, especially when the Aggregate structure grows beyond just a couple of Entities, it is cleaner to react on events being published in other Entities of the same Aggregate (multi-entity Aggregates are explained in more detail xref:modeling/multi-entity-aggregates.adoc[here]).
-However, since the Event Handling methods are also invoked when reconstructing Aggregate state, special precautions must be taken.
+=== Entity member relationships
 
-It is possible to `apply()` new events inside an Event Sourcing Handler method.
-This makes it possible for an Entity 'B' to apply an event in reaction to Entity 'A' doing something.
-Axon will ignore the `apply()`invocation when replaying historic events upon sourcing the given Aggregate. Do note that in the scenario where Event Messages are published from an Event Sourcing Handler, the Event of the inner `apply()` invocation is only published to the entities after all entities have received the first event.
-If more events need to be published, based on the state of an entity after applying an inner event, use `apply(...).andThenApply(...)`.
+In some cases, especially when the entity structure grows beyond a single entity, it is useful to compose entities with child entities.
+Child entities can handle commands and events just like root entities (multi-entity structures are explained in more detail xref:modeling/multi-entity-command-models.adoc[here]).
+
+Child entities are annotated with `@EntityMember`:
+
+[source,java]
+----
+import org.axonframework.modelling.entity.annotation.EntityMember;
+
+public class Project {
+
+    private final String projectId;
+
+    @EntityMember
+    private List<Developer> developers = new ArrayList<>();
+
+    @EntityMember
+    private List<Milestone> milestones = new ArrayList<>();
+
+    // command handlers and event sourcing handlers
+}
+----
 
 [NOTE]
-.Reacting to external events
+.Event sourcing scope
+====
+An Entity processes events that belong to its consistency boundary.
+With the introduction of Dynamic Consistency Boundary (DCB), this boundary can span multiple event streams.
+Event sourcing handlers are used to recreate the state of the Entity from its events.
 ====
-An Aggregate *cannot* handle events from other sources then itself.
-This is intentional as the Event Sourcing Handlers are used to recreate the state of the Aggregate.
-For this, it only needs its own events as those represent its state changes.
 
 To make an Aggregate react to events from other Aggregate instances, xref:sagas:index.adoc[Sagas] or xref:events:event-handlers.adoc[Event Handling Components] should be leveraged
```

Comment:
everything below here seems to be legacy aggregate documentation and should be removed